packages/create-expo-app/src/analytics.ts

```diff
@@ -0,0 +1,189 @@
+import JsonFile from '@expo/json-file';
+import crypto from 'crypto';
+import fs from 'fs';
+import fetch from 'node-fetch';
+import os from 'os';
+
+import { getStateJsonPath } from './paths';
+import { getSession } from './sessionStorage';
+
+const packageJSON = require('../package.json');
+
+const xdlWriteKey = '1wabJGd5IiuF9Q8SGlcI90v8WTs';
+const analyticsEndpoint = 'https://cdp.expo.dev/v1/batch';
+const version = '1.0.0';
+const library = 'create-expo-app';
+
+//#region mostly copied from @expo/rudder-sdk-node https://github.com/expo/rudder-sdk-node/blob/main/index.ts
+// some changes include:
+// - the identity being injected inside of the enqueue method, rather than as a function argument.
+// - a global event queue that gets cleared after each flush
+// - using node's crypto library for hashing and uuidv4
+type AnalyticsPayload = {
+  messageId: string;
+  _metadata: any;
+  context: any;
+  type: string;
+  originalTimestamp: Date;
+  [key: string]: any;
+};
+type AnalyticsMessage = {
+  context?: {
+    [key: string]: unknown;
+  };
+  integrations?: {
+    [destination: string]: boolean;
+  };
+  properties?: {
+    [key: string]: unknown;
+  };
+  timestamp?: Date;
+  [key: string]: unknown;
+};
+type AnalyticsIdentity =
+  | {
+      userId: string;
+    }
+  | {
+      userId?: string;
+      anonymousId: string;
+    };
+
+const messageBatch = [] as AnalyticsPayload[];
+
+let analyticsIdentity: AnalyticsIdentity | null = null;
+
+// call before tracking any analytics events.
+// if track/identify are called before this method they will be dropped
+export async function initializeAnalyticsIdentityAsync() {
+  if (analyticsIdentity) {
+    return;
+  }
+  analyticsIdentity = await getAnalyticsIdentityAsync();
+}
+
+export function identify() {
+  enqueue('identify', {});
+}
+
+export function track(
+  message: AnalyticsMessage & {
+    event: string;
+  }
+) {
+  enqueue('track', { ...message, context: getAnalyticsContext() });
+}
+
+function enqueue(type: 'identify' | 'track', message: any) {
+  if (!analyticsIdentity) {
+    // do not send messages without identities to our backend
+    return;
+  }
+
+  message = { ...message, ...analyticsIdentity };
+  message.type = type;
+
+  if (message.type === 'identify') {
+    message.traits ??= {};
+    message.context ??= {};
+    message.context.traits = message.traits;
+  }
+
+  message.context = {
+    library: {
+      name: `@expo/expo-cli/${library}`,
```

Comment:
?

Reply:
i wasn't sure what you were referring to but did shorten the string. lmk if it's ok